src/rules/no-private-properties.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * @fileoverview Disallows usages of "non-public" property bindings
+ * @author Michael Stramel <https://github.com/stramel>
+ */
+
+import {Rule} from 'eslint';
+import * as ESTree from 'estree';
+import {TemplateAnalyzer} from '../template-analyzer';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const rule: Rule.RuleModule = {
+  meta: {
+    docs: {
+      description: 'Disallows usages of "non-public" property bindings',
+      category: 'Best Practices',
+      url:
+        'https://github.com/43081j/eslint-plugin-lit/blob/master/docs/rules/no-private-properties.md'
+    },
+    messages: {
+      unsupported: 'TODO'
+    },
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          private: {type: 'string', minLength: 1, format: 'regex'},
+          protected: {type: 'string', minLength: 1, format: 'regex'}
+        },
+        additionalProperties: false,
+        minProperties: 1
+      }
+    ]
+  },
+
+  create(context): Rule.RuleListener {
+    // variables should be defined here
+    const config: {private?: string; protected?: string} =
```

Comment:
a `Partial<{private: string; protected: string}>` is preferred so we don't overmatch